### PR TITLE
[Merged by Bors] - chore: adaptations for nightly-2024-09-08

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

File: Mathlib/Data/List/Basic.lean [hunk 1]
```diff
@@ -947,13 +940,6 @@ theorem infix_bind_of_mem {a : α} {as : List α} (h : a ∈ as) (f : α → Lis
 theorem map_eq_map {α β} (f : α → β) (l : List α) : f <$> l = map f l :=
   rfl
 
-#adaptation_note
-/--
-`nolint simpNF` should be removed after nightly-2024-09-07.
--/
-@[simp, nolint simpNF]
-theorem map_tail (f : α → β) (l) : map f (tail l) = tail (map f l) := by cases l <;> rfl
-
```
Comment (semorrison) [inline Mathlib/Data/List/Basic.lean:956]: This was upstreamed so can be removed entirely.

File: Mathlib/Data/List/Basic.lean [hunk 2]
```diff
@@ -2227,14 +2213,6 @@ end Forall
 
 /-! ### Miscellaneous lemmas -/
 
-@[simp]
-theorem getElem_attach (L : List α) (i : Nat) (h : i < L.attach.length) :
```
Comment (semorrison) [inline Mathlib/Data/List/Basic.lean:2231]: Upstreamed, without the `.1`. Seems to work fine.

File: Mathlib/Analysis/Calculus/Deriv/Basic.lean [hunk 3]
```diff
@@ -416,7 +416,7 @@ theorem norm_deriv_eq_norm_fderiv : ‖deriv f x‖ = ‖fderiv 𝕜 f x‖ := b
 
 theorem DifferentiableAt.derivWithin (h : DifferentiableAt 𝕜 f x) (hxs : UniqueDiffWithinAt 𝕜 s x) :
     derivWithin f s x = deriv f x := by
-  unfold derivWithin deriv
+  unfold _root_.derivWithin deriv
```
Comment (semorrison) [inline Mathlib/Analysis/Calculus/Deriv/Basic.lean:419]: `unfold` now errors if the name is ambiguous.

**Review by TwoFX — APPROVED** [review]
Thanks!
bors r+

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Pull request closed without merging**